main.tf

```diff
@@ -41,178 +34,132 @@ locals {
     backend k3s_nodes
         mode tcp
         balance roundrobin
-        server k3s_node1 ${crusoe_compute_instance.k3s_0.network_interfaces[0].private_ipv4.address}:6443 check
-        server k3s_node2 ${crusoe_compute_instance.k3s_1.network_interfaces[0].private_ipv4.address}:6443 check
-        server k3s_node3 ${crusoe_compute_instance.k3s_2.network_interfaces[0].private_ipv4.address}:6443 check
-  EOT
-}
-
-resource "local_file" "haproxy_config" {
-  filename = "haproxy.cfg"
-  content  = local.haproxy_local
+        %{~ for index, instance in crusoe_compute_instance.k3s_headnode ~}
+        server k3s_node${index} ${instance.network_interfaces[0].private_ipv4.address}:6443 check
+        %{~ endfor ~}
+    EOT
+  headnode_entry    = local.use_lb ? one(crusoe_compute_instance.k3s_lb) : one(crusoe_compute_instance.k3s_headnode)
+  ingress_interface = local.headnode_entry.network_interfaces[0]
+  headnode_has_gpu  = strcontains(var.headnode_instance_type, "sxm-ib")
```

Comment:
Is this future proof? Is it a reasonable assumption that the instance type will always have an SXM-connected GPU? Or do we foresee GPUs being supported in future which have a different connection type?